### ISSUE TITLE
After a certain period of time, it returns null when reading.

### ISSUE THREAD

**Issue by hasanuc20** [cabbi/modbus_client_tcp#7]
There is no connection loss with my Modbus device. I am still connected to the device.After some time has passed, the code I wrote for you returns null to me and I constantly receive this information. why could it be.
 Future<String> readHoldingRegister(int adres) async {
    try {
      var value = ModbusInt16Register(
        name: 'name',
        address: adres,
        type: ModbusElementType.holdingRegister,
      );
      var valueRequest = value.getReadRequest();
      final responce = await modbusclient.send(value.getReadRequest(
        unitId: 1,
        responseTimeout: const Duration(seconds: 5),
      ));
      if (ModbusResponseCode.requestSucceed == responce) {
        if (valueRequest.element.value != null) {
          return (valueRequest.element.value.truncate()).toString();
        } else {
          return 'null.';
        }
      } else {
      return 'null';
      }
    } on SocketException catch (e) {
      print('SocketException caught: $e');
      rethrow; // Re-throw the exception to propagate it to the caller.
    }
  }

**Comment by cabbi** [comment]
Please check the response value returned by the `send` method. This will tell you more about the failure reason.

**Comment by hasanuc20** [comment]
ModbusResponseCode.requestRxFailed It rotates.

**Comment by cabbi** [comment]
what are logs showing?

**Comment by hasanuc20** [comment]
I looked at the error logs. Nothing is visible, everything is normal. I am reading the data at a speed of 100 ms. This may be affecting.

**Comment by cabbi** [comment]
Since you are making a normal register request, the following lines of code are the only two cases where the Modbus Client library returns `ModbusResponseCode.requestRxFailed`
 ```
     if (transactionId != resView.getUint16(0)) {
        ModbusAppLogger.warning("Invalid TCP transaction id",
            "$transactionId != ${resView.getUint16(0)}");
        _timeout.complete();
        request.setResponseCode(ModbusResponseCode.requestRxFailed);
        return;
      }
      if (0 != resView.getUint16(2)) {
        ModbusAppLogger.warning(
            "Invalid TCP transaction id", "${resView.getUint16(2) != 0}");
        _timeout.complete();
        request.setResponseCode(ModbusResponseCode.requestRxFailed);
        return;
      }
```
As you can see, in both cases you have a warning log you might see by enabling logging `ModbusAppLogger(Level.FINE);
`.

Which modbus device are you connected too?

**Comment by hasanuc20** [comment]
It works for a certain period of time. All the codes I read have the same problem. I couldn't figure it out. error2 is returned.
Future<String> readHoldingRegister(int adres) async {
    try {
      var value = ModbusInt16Register(
        name: 'name',
        address: adres,
        type: ModbusElementType.holdingRegister,
      );
      var valueRequest = value.getReadRequest();
      final responce = await modbusclient.send(value.getReadRequest(
        unitId: 1,
        responseTimeout: const Duration(seconds: 10),
      ));
      if (ModbusResponseCode.requestSucceed == responce) {
        if (valueRequest.element.value != null) {
          return (valueRequest.element.value.truncate()).toString();
        } else {
          return 'error1';
        }
      } else {
        return 'error2';
      }
    } on SocketException catch (e) {
      print('SocketException caught: $e');
      rethrow; // Re-throw the exception to propagate it to the caller.
    }
  }

**Comment by hasanuc20** [comment]
Future<bool> readCoil(int adres) async {
    try {
      var coil = ModbusCoil(
        name: "name",
        address: adres,
      );
      var coilRequest =
          coil.getReadRequest(unitId: 1, responseTimeout: const Duration(seconds: 10));
      var response = await modbusclient.send(coilRequest);
      if (ModbusResponseCode.requestSucceed == response) {
        return coilRequest.element.value;
      } else {
        print('error2');
        return false;
      }
    } on SocketException catch (e) {
      print('SocketException yakalandı: $e');
      return false;
    }
  }In this code it returns error2.I am connecting to schneider plc tcp.
![Ekran görüntüsü 2024-05-13 171406](https://github.com/cabbi/modbus_client_tcp/assets/135219553/40410ed7-28b4-48e5-a48a-58dd95eea71c)

**Comment by cabbi** [comment]
Hi, 

as I was mentioning in a reply here above, you might enable logs by putting this line at beginning of your program `ModbusAppLogger(Level.FINE);`

Then let me know what is the log showing you.

**Comment by hasanuc20** [comment]
I did what you said. I am sharing the screenshot.
![Ekran görüntüsü 2024-05-14 102941](https://github.com/cabbi/modbus_client_tcp/assets/135219553/857e7459-b76b-4f2f-8333-7a978dda6f44)

**Comment by cabbi** [comment]
Ok, you've found a bug. I've published a fix in new modbus_client_tcp 1.2.1 package. 

Please upgrade your modbus_client_tcp  reference to latest and close this issue if fixed, thanks.
